sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -6,15 +6,2076 @@
+
+        tx_power_flag_dict = {'tx_power_high_alarm': tx_power_high_alarm_dict,
+                              'tx_power_low_alarm': tx_power_low_alarm_dict,
+                              'tx_power_high_warn': tx_power_high_warn_dict,
+                              'tx_power_low_warn': tx_power_low_warn_dict,}
+        return tx_power_flag_dict
+
+    def get_tx_bias_flag(self):
+        '''
+        This function returns TX bias out of range flag on TX media lane
+        '''
+        tx_bias_high_alarm = self.xcvr_eeprom.read(consts.TX_BIAS_HIGH_ALARM_FLAG)
+        tx_bias_low_alarm = self.xcvr_eeprom.read(consts.TX_BIAS_LOW_ALARM_FLAG)
+        tx_bias_high_warn = self.xcvr_eeprom.read(consts.TX_BIAS_HIGH_WARN_FLAG)
+        tx_bias_low_warn = self.xcvr_eeprom.read(consts.TX_BIAS_LOW_WARN_FLAG)
+        tx_bias_high_alarm_dict = dict()
+        tx_bias_low_alarm_dict = dict()
+        tx_bias_high_warn_dict = dict()
+        tx_bias_low_warn_dict = dict()
+        for bitpos in range(BYTELENGTH):
+            tx_bias_high_alarm_dict['TX_lane%d' %(bitpos+1)] = bool((tx_bias_high_alarm >> bitpos) & 0x1)
+            tx_bias_low_alarm_dict['TX_lane%d' %(bitpos+1)] = bool((tx_bias_low_alarm >> bitpos) & 0x1)
+            tx_bias_high_warn_dict['TX_lane%d' %(bitpos+1)] = bool((tx_bias_high_warn >> bitpos) & 0x1)
+            tx_bias_low_warn_dict['TX_lane%d' %(bitpos+1)] = bool((tx_bias_low_warn >> bitpos) & 0x1)
+
+        tx_bias_flag_dict = {'tx_bias_high_alarm': tx_bias_high_alarm_dict,
+                              'tx_bias_low_alarm': tx_bias_low_alarm_dict,
+                              'tx_bias_high_warn': tx_bias_high_warn_dict,
+                              'tx_bias_low_warn': tx_bias_low_warn_dict,}
+        return tx_bias_flag_dict
+
+    def get_rx_power_flag(self):
+        '''
+        This function returns RX power out of range flag on RX media lane
+        '''
+        rx_power_high_alarm = self.xcvr_eeprom.read(consts.RX_POWER_HIGH_ALARM_FLAG)
+        rx_power_low_alarm = self.xcvr_eeprom.read(consts.RX_POWER_LOW_ALARM_FLAG)
+        rx_power_high_warn = self.xcvr_eeprom.read(consts.RX_POWER_HIGH_WARN_FLAG)
+        rx_power_low_warn = self.xcvr_eeprom.read(consts.RX_POWER_LOW_WARN_FLAG)
+        rx_power_high_alarm_dict = dict()
+        rx_power_low_alarm_dict = dict()
+        rx_power_high_warn_dict = dict()
+        rx_power_low_warn_dict = dict()
+        for bitpos in range(BYTELENGTH):
+            rx_power_high_alarm_dict['RX_lane%d' %(bitpos+1)] = bool((rx_power_high_alarm >> bitpos) & 0x1)
+            rx_power_low_alarm_dict['RX_lane%d' %(bitpos+1)] = bool((rx_power_low_alarm >> bitpos) & 0x1)
+            rx_power_high_warn_dict['RX_lane%d' %(bitpos+1)] = bool((rx_power_high_warn >> bitpos) & 0x1)
+            rx_power_low_warn_dict['RX_lane%d' %(bitpos+1)] = bool((rx_power_low_warn >> bitpos) & 0x1)
+
+        rx_power_flag_dict = {'rx_power_high_alarm': rx_power_high_alarm_dict,
```

Comment:
can you rename the dict keys similar to the ones defined in sff8436?.

Reply:
please check whether the dict keys defined are similar to existing sff8436/sff8636?